docs/roles.md

```diff
@@ -0,0 +1,151 @@
+- Perform Transaction Request - `POST /transactionRequests`
+- Retrieve Quote Information - `GET /quotes/{ID}`
+- Calculate Quote - `POST /quotes`
+- Retrieve Bulk Quote Information - `GET /bulkQuotes/{ID}`
+- Calculate Bulk Quote - `POST /bulkQuotes`
+- Perform Authorization - `GET /authorizations/{ID}`
+- Retrieve Transfer Information - `GET /transfers/{ID}`
+- Perform Transfer - `POST /transfers`
+- Retrieve Bulk Transfer Information - `GET /bulkTransfers/{ID}`
+- Perform Bulk Transfer - `POST /bulkTransfers`
+- Retrieve Transaction Information - `GET /transactions/{ID}`
+- Update Authorization Request - `PUT /authorizationRequests/{ID}` _[new pisp call]_
+
+**Inbound _(Note: some ommited for brevity)_**
+- Receive Authorization Creation Request - `POST /authorizationRequests/{ID}`
+
+
+### 2. PISP:
+**Outbound**
+- Lookup Party Information - `GET /parties/{Type}/{ID}, /parties/{Type}/{ID}/{SubId}`
+- Perform Authorization on this Thing `POST /authorizations/{ID}` [newly added for PISP case]
+- Update Authorization - `PUT /authorizations/{ID}`
+- Create Consent Request - `POST /consentRequests/{ID}`
+- Update Consent Request - `PUT /consentRequests/{ID}`
+- Update Consent - `POST /consents/{ID}`
+- Return Quote Information - `PUT /quotes/{ID}` _[this probably won't be needed since the quote is going to be a part of the authorizations callback]_
+- Return Quote Information Error - `PUT /quotes/{ID}/error` _[same as above]_
+- Get 3rd party request information - `GET /thirdPartyRequest/transaction/{ID}`
+- Initiate 3rd party request - `POST /thirdPartyRequest/transaction/{ID}`
+
+**Inbound**
+- Return Party Information - `PUT /parties/{Type}/{ID}, /parties/{Type}/{ID}/{SubId}`
+- Return Party Information Error - `PUT /parties/{Type}/{ID}/error, /parties/{Type}/{ID}/{SubId}/error`
+- Return Authorization Result - `PUT /authorizations/{ID}`
+- Return Authorization Error - `PUT /authorizations/{ID}/error`
+- Update Consent Request - `PUT /consentRequests/{ID`
+- Update Consent Request Error - `PUT /consentRequests/{ID}/error`
+- Update Consent - `POST /consents/{ID}`
+- Update 3rd Party Transaction Request - `PUT /thirdPartyRequest/transaction/{ID}`
+- Update 3rd Party Transaction Request Error - `PUT /thirdPartyRequest/transaction/{ID}/error`
+
+## Questions:
+
+- What endpoint are we using to get account balances/metadata? Should this be `GET /parties...` or something else, such as `thirdPartyRequests/balance`?
+- In the FSPIOP API Definition v1.1, `/authorizations` resource is used solely in conjunction with internal transaction requests. With the addition of a new PISP transaction requests `/thirdPartyRequests/transaction`, we may need to reconsider how `/authorizations` is defined so that it covers both the 'internal' (or existing) transaction request scenario, as well as the new 'Third Party' scenario
+
+## Performance Considerations:
+
+We assume Role Based Access Controls will be implemented at an API Gateway.
```

Comment:
Is it possible that RBAC could be used not only on APIGateway but also in endpoints implementations to deliver different behavior for different types of Parties (PISP and DFSP)? If yes how the information about PartyType will be passed down?

Reply:
Maybe it is possible to implement RBAC on the SDK Scheme Adapter instead of APIGateway? 
https://github.com/mojaloop/sdk-scheme-adapter/ 
It is placed as a protocol adapter between Hub and Parties (DFSP and maybe PISP...)
It is dedicated to performing the operations related to security and header processing.

Reply:
The plan is, I think, that the scheme will maintain a central register of which services each participant offers. and this register will be used to implement the RBAC test. We envisage that this will be slowly changing information, and that it will not therefore be necessary for participants to state on each message what service they are requesting. I believe that we use fixed IP addresses, so the RBAC should be able to identify the party directly.

Unfortunately, we don't mandate that participants should make use of the scheme adapter, so it would not be a reliable location for the test.

Reply:
thank you @MichaelJBRichards for the explanation. But maybe it would be worth investigating the need for an adapter for PISP?

Reply:
@eoln to Michael's point, I think we can safely assume that these access controls lie outside of the scope for what we are doing for the moment. 

As for the adapter for PISP, can you expand a little more on this? Do you mean a sdk-scheme-adapter for PISPs instead of DFSPs?